tests/unit/utils/test_ratelimiter.py

```diff
@@ -0,0 +1,110 @@
+import time
+from datetime import datetime
+from unittest.mock import AsyncMock, patch
+
+import pytest
+from freezegun import freeze_time
+
+from deepset_cloud_sdk._utils.ratelimiter import RateLimiter
+
+pytest_plugins = ("pytest_asyncio",)
+
+
+# These tests need timeouts, if there is a bug in RateLimiter, it could result in the tests running forever
+@pytest.mark.timeout(0.1)
+class TestRateLimiter:
```

Comment:
If you make the rate limit a param of the constructor with the current value as default, then this would in my opinion ease unit testing a lot, no? Currently this is almost tautological testing as you manipulate a of class internals